doc/scratch-space.md

```diff
@@ -0,0 +1,10 @@
+# CDI Scratch space
+Containerized Data Importer(CDI) requires scratch space for certain operations to complete, this temporary space needs to be obtained from somewhere. Kubernetes has some options available to get temporary space like emptyDir volumes, however that space is shared among pods and it is uncertain how much space is available or what the node behavior will be if CDI fills up that space with a large image. For this and other reasons CDI will create scratch space from available PVs using a storage class. This scratch space will then be used to process the data before writing it to the target PVC.
+
+CDI uses the following mechanism to determine which storage class to use:
+
+1. Read the CDI config field _scratchSpaceStorageClass_ if that field exists, and the value matches one of the storage classes in the cluster, it will be used to create scratch space.
+2. If 1 is empty or didn't match a storage class, look up the _default_ storage class in the cluster, if that exists use that storage class.
+3. If there is no default storage class in the cluster, then use the storage class of the PersistentVolumeClaim(PVC) that is backing the DataVolume(DV) that started the CDI operation.
+
+If none of those exist, then CDI will be unable to create scratch space. This means that none of the operations that require scratch space will work, however operations that do not require scratch space will continue to operate normally.
```

Comment:
Can we specify explicitly which operations require scratch and which are not?

Reply:
We can, that will probably help @aglitke with his task to document all the valid permutations of formats we support. Let me update the PR to include a grid with operations that require scratch space.

Reply:
Updated.